plutus-contract/src/Wallet/Emulator/Wallet.hs

```diff
@@ -268,7 +268,7 @@ validateTxAndAddFees feeCfg slotCfg ownTxOuts utx = do
     -- Balance and sign just for validation
     tx <- handleBalanceTx ownTxOuts utx
     signedTx <- handleAddSignature tx
-    let utxoIndex        = Ledger.UtxoIndex $ unBalancedTxUtxoIndex utx <> (toTxOut <$> ownTxOuts)
+    let utxoIndex        = Ledger.UtxoIndex $ fmap toTxOut $ (U.fromScriptOutput <$> unBalancedTxUtxoIndex utx) <> ownTxOuts
```

Comment:
@j-mueller After thinking some more this here seems to be the clue that there is no failure. The PubKeys are only unavailable in the lookups. But the wallet knows how to lookup its own outputs (`ownTxOuts` here) so `getUnspentOutput` just works!

Reply:
`getUnspentOutput` works, and it gives you an unspent output from the wallet. But there is no guarantee that the wallet selects the same unspent output when `balanceTx` is called a second time, for the transaction that creates the token by spending this specific output. 

The UTXO could have been spent in the meantime, or maybe the wallet decides that there is a different collection of UTXOs that is better suited in the second call to `balanceTx`.

Reply:
Because of this, I'm inclined to think that we have to go back to creating a script output (with the PubKey contract) in a separate transaction before we can set up the state machine. What do you think @sjoerdvisscher  @silky  ?

Reply:
@j-mueller I agree, but that’s a separate issue I think? So we can merge this one?

Reply:
Yes, let's merge it!